src/category_theory/groupoid/basic.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2022 Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémi Bottinelli
+-/
+import category_theory.groupoid
+import category_theory.is_connected
+import combinatorics.quiver.basic
+
+/-!
+This file defines a few basic properties of groupoids.
+-/
+
+namespace category_theory
+
+namespace groupoid
+
+variables (C : Type*) [groupoid C]
+
+section thin
+
+lemma is_thin_iff : (quiver.is_thin C) ↔ (∀ (c : C), subsingleton (c ⟶ c)) :=
+begin
+  refine ⟨λ h c, h c c, λ h c d, subsingleton.intro $ λ f g, _⟩,
+  haveI := h d,
+  calc f = f ≫ (inv g ≫ g) : by simp only [inv_eq_inv, is_iso.inv_hom_id, category.comp_id]
+     ... = f ≫ (inv f ≫ g) : by congr
+     ... = g               : by simp only [inv_eq_inv, is_iso.hom_inv_id_assoc],
+end
+
+end thin
+
+section disconnected
+
+/-- A subgroupoid is disconnected if it only has loops. -/
+def is_disconnected := ∀ (c d : C), (c ⟶ d) → c = d
```

Comment:
I think this name might be confused with the negation of `category_theory.is_connected`. Maybe `is_totally_disconnected`?

This can also be generalized into quivers, but I don't really have an application for this so I don't think it is worth doing now.